mirrord/cli/src/config.rs

```diff
@@ -399,7 +399,16 @@ pub(super) struct PortForwardArgs {
 #[derive(Clone, Debug, PartialEq)]
 pub struct PortMapping {
     pub local: SocketAddr,
-    pub remote: SocketAddr,
+    pub remote: (RemoteAddr, u16),
+}
+
+#[derive(Clone, Debug, PartialEq)]
+pub enum RemoteAddr {
+    // if the remote is given as an IPv4
+    Ip(Ipv4Addr),
+    // if the remote needs DNS resolution, we'll delay until PortForwarder can use the
+    // AgentConnection
+    Hostname(String),
 }
```

Comment:
At this point I'd extend the doc for `PortForwardArgs::port_mappings` and describe what an argument should look like and what it does and that hostnames are resolved remotely in the cluster (the docs are visible to the user when they do `mirrord port-forward --help`)